### PR TITLE
Fix/merge issues

### PR DESCRIPTION
**Pull request by MylesNottingham**
This will fix issues caused by the two-branch merge, organize methods & tests, add game_spec, harden current tests, and refactor for convention.

☑ RSpec
☑ SimpleCov
☑ spec_harness
☑ rubocop

### CODE REVIEW

**Review by javenb022 — APPROVED** [review]
Lovin it

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -84,35 +84,12 @@ def worst_offense
...
-    end[0]
-
-    get_team_name(least_accurate_team_id)
-  end
-
   def highest_scoring_visitor
     away_pts_per_game = Hash.new(0)
     total_goals_by_away_team.each do |team_id, goals|
-      away_pts_per_game[team_id] = (goals / total_games_played_by_team[team_id])
+      away_pts_per_game[team_id] = (goals / total_games_played_by_team_in_games[team_id])
```
Comment (garrettgregor) [inline lib/stat_tracker.rb:90]: This looks super clean now! Thank you!

**Review by garrettgregor — APPROVED** [review]
Thanks for refactoring some of those methods and adding those additional tests!

**Comment by garrettgregor** [comment]
🙌🏻 Well done! Thanks for adding that missing test too!

**Pull request merged**